app/src/main/java/com/eveningoutpost/dexdrip/Home.java

```diff
@@ -2628,7 +2635,7 @@ private void updateCurrentBgInfoCommon(DexCollectionType collector, TextView not
                     dialog = builder.create();
                     dialog.show();
                 } else {
-                    if (!Experience.gotData() && !QuickSettingsDialogs.isDialogShowing() && JoH.ratelimit("start-sensor_prompt", 20)) {
+                    if (!Experience.gotData() && !QuickSettingsDialogs.isDialogShowing() && JoH.ratelimit("start-sensor_prompt", 20) && !notConnectedToG6Yet) {
```

Comment:
So your `!notConnectedToG6Yet` condition should come before the ratelimit as everything to the right of the ratelimit is not evaluated when the ratelimit has been exceeded. I believe in this case you would only want the ratelimit to restricting firing of the conditional block after all conditions have been met.

Reply:
In other words, if the user has not selected G6 or if they have and it is already connected, the dialog will be shown as it did before this PR.
However, if G6 is selected and is not connected yet, this PR does not allow the dialog to be shown.

Reply:
Yes but if the user reaches the condition and the timekeeper condition is not met, but then the connection occurs in the background and would be true and the user then re-enters this conditional block again within the 20 second rate limit period, it will fail because the ratelimit has been triggered even though it otherwise would have passed if the check order is changed so the ratelimiting is last. The `&&` operator means the other clauses are not evaluated once one is false. Simply checking the ratelimit by calling its method triggers the time to be stored so that the condition will fail the next time you call it within the specified time period.

Reply:
I'm so sorry I don't follow you.
I keep comparing what xDrip does now to what I want it to do after this PR, to help me see that this PR is correct.

If ratelimit fails, xDrip does not show the dialog right now.  In that case, I don't want xDrip to show the dialog after this PR either.  So, I don't understand the dellima you are describing.

Reply:
Sorry!
I misunderstood the purpose of the bracket.  As soon as ratelimit executes, it is going to show the dialog.
Thanks for catching this.

Reply:
Done

Reply:
Tested